services/wallet/activity/service.go

```diff
@@ -74,7 +74,7 @@ type FilterResponse struct {
 // FilterActivityAsync allows only one filter task to run at a time
 // and it cancels the current one if a new one is started
 // All calls will trigger an EventActivityFilteringDone event with the result of the filtering
-func (s *Service) FilterActivityAsync(ctx context.Context, addresses []common.Address, chainIDs []w_common.ChainID, filter Filter, offset int, limit int) {
+func (s *Service) FilterActivityAsync(requestID int, addresses []common.Address, chainIDs []w_common.ChainID, filter Filter, offset int, limit int) {
 	s.scheduler.Enqueue(filterTask, func(ctx context.Context) (interface{}, error) {
```

Comment:
IMO we want async calls from separate requestIDs not to affect each other. If not, a request from the Send modal might cancel a previous request from the activity tab, right?

Reply:
Right, I missed this; thanks for the heads up.